flagsmith-engine/segments/models.ts

```diff
@@ -63,7 +63,9 @@ export class SegmentConditionModel {
                 return !traitValue.includes(this.value);
             },
             evaluateRegex: (traitValue: any) => {
-                return !!traitValue.match(new RegExp(this.value));
+                if (this.value != null){
```

Comment:
We should probably update this to: 

```javascript
return !!this.value && !!traitValue.match(new RegExp(this.value))
```

At the moment, your code will return `undefined` if `this.value == null` - we should ideally make sure it always returns a boolean value. Likely it wouldn't make much difference since `undefined` would behave very much like `false` anyway but it's cleaner to handle it this way.